### ISSUE TITLE
Twitter sentiment analysis

### ISSUE THREAD

**Issue by Yuvika-14** [abhisheks008/ML-Crate#495]
### ML-Crate Repository (Proposing new issue)
Project Title :Twitter sentiment analysis
Aim :to predict whether a comment is positive or negative on twitter.
Dataset:https://www.kaggle.com/datasets/kazanova/sentiment140
 Approach:ensemble methods, gradient boosting, neural networks.
Take care of the missing data if there are any categorical values then will use one hot encoder or label encoder depending upon the issue.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by hemant933** [comment]
Full name :Hemant chaudhary
GitHub Profile Link : github.com/hemant933
Participant ID (If not, then put NA) :
Approach for this Project : first we need to process the data to handle any missing data , than apply models specified by using vectorization and pipeling .
What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.) IWOC 

so , plz assign it to me

**Comment by abhisheks008** [comment]
Hi @Yuvika-14 this project is already present in this project repo, https://github.com/abhisheks008/ML-Crate/tree/main/Sentimental%20Analysis%20of%20tweets.

If you wanna enhance this project then you can share your approach.

**Comment by JagritiGautam793** [comment]
Full name :Jagriti Gautam
GitHub Profile Link : https://github.com/JagritiGautam793
Participant ID (If not, then put NA) :
Approach for this Project :After processing of the data i will try for the real time analysis of the tweets by using twitter api (Tweepy)and after  then applying nlp and pretrained transformer model after that visualizing the data through donut charts ..

What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.) IWOC

so , plz assign it to me

**Comment by JagritiGautam793** [comment]
Can u assign me @abhisheks008

**Comment by abhisheks008** [comment]
Please check the previous comments.

**Comment by JagritiGautam793** [comment]
Sir i will improve this model by using Roberta Model(from hugging face) rather than vader ....This transformer model account for words but also the context related to words .As human language depend more on context ... Vader is not that much accurate in analyzing the context ..
Sir pls if u can assign me and guide through it

**Comment by abhisheks008** [comment]
Cool, assigned to you @JagritiGautam793

**Comment by abhisheks008** [comment]
Unassigned as the open source event ended up.

**Comment by JagritiGautam793** [comment]
Sir i am almost done .. Was the event date not upto 15 th

**Comment by abhisheks008** [comment]
No @JagritiGautam793 IWOC 2024 deadline was Feb 11th, 2024 23:59 hours.

**Comment by abhisheks008** [comment]
This issue is not being assigned to you as the program has already completed, hence the assignment has been removed.

**Comment by abhisheks008** [comment]
@JagritiGautam793

**Comment by shivansh-2003** [comment]
Can You Please Assign this issue under SSOC. 2024 Season 3
Shivansh Mahajan
Github:- https://github.com/shivansh-2003
Participation ID:- NA
I will make a pickle file of the Fitted ML model and then deploy it to Streamlit Python Framework to display the feature and predict the Risk Assessment Of Vehicle . I am well versed with making Streamlit App you can check out my linkedin :-https://www.linkedin.com/in/shivansh-mahajan-13227824a/ and Git repository .
My some recent Project in NLP Projects https://www.linkedin.com/feed/update/urn:li:activity:7199784822737682432/(SPAM classifier)
https://www.linkedin.com/feed/update/urn:li:activity:7201206409605091328/ . (NLP APP)
can u assign me with this issue @abhisheks008
Participation Role:- SSOC Season 3

**Comment by abhisheks008** [comment]
Contributions will start from June 1, 2024. Till then please have some patience.

**Comment by Tanishka023** [comment]
Hey @abhisheks008 Sir, please assign this issue to me under SSOC'24, I really want to work on this.

**Comment by Tanishka023** [comment]
Full name : Tanishka Bhalla
GitHub Profile Link: https://github.com/Tanishka023
Participant ID (If not, then put NA) :NA
Approach for this Project : Extract data from Twitter API -> Processing the data -> Feature Extraction -> NB/CNN/LR -> Model Training -> Evaluation 
What is your participant role? SSoC Season 3

**Comment by abhisheks008** [comment]
> Full name : Tanishka Bhalla GitHub Profile Link: https://github.com/Tanishka023 Participant ID (If not, then put NA) :NA Approach for this Project : Extract data from Twitter API -> Processing the data -> Feature Extraction -> NB/CNN/LR -> Model Training -> Evaluation What is your participant role? SSoC Season 3

Can you implement 3-4 models for this project?

**Comment by Tanishka023** [comment]
> > Full name : Tanishka Bhalla GitHub Profile Link: https://github.com/Tanishka023 Participant ID (If not, then put NA) :NA Approach for this Project : Extract data from Twitter API -> Processing the data -> Feature Extraction -> NB/CNN/LR -> Model Training -> Evaluation What is your participant role? SSoC Season 3
> 
> Can you implement 3-4 models for this project?

Yes! @abhisheks008

**Comment by abhisheks008** [comment]
> > > Full name : Tanishka Bhalla GitHub Profile Link: https://github.com/Tanishka023 Participant ID (If not, then put NA) :NA Approach for this Project : Extract data from Twitter API -> Processing the data -> Feature Extraction -> NB/CNN/LR -> Model Training -> Evaluation What is your participant role? SSoC Season 3
> > 
> > 
> > Can you implement 3-4 models for this project?
> 
> Yes! @abhisheks008

One issue at a time.

**Comment by shivamkrishna1000** [comment]
Full Name : Shivam Krishna
GitHub Profile Link : https://github.com/shivamkrishna1000
Participant ID : NA
Approach for this Project : First we will handle the missing values if any -> use label encoding if needed -> use TfidfVectorizer for feature extraction -> Use various models such as XGB/Random Forest Classification/LR/SVM/Gradient Boosting Classifier for model training -> Model evaluation based on accuracy score.
What is your participant role? SSoC Season 3

Please assign me this issue.

**Comment by why-aditi** [comment]
Aditi Kala
Github:- https://github.com/why-aditi
Participation ID:- NA
Approach: Text Cleaning: Remove unnecessary elements from the tweets such as: URLs, Hashtags, Punctuation, Numbers, Special characters, Remove common words that do not contribute much to the sentiment
Tokenization: Split the text into individual words or tokens.
Sentiment analysis: VADER
Extract sentiment scores or labels (e.g., positive, negative, neutral)
Participation Role:- SSOC Season 3

**Comment by abhisheks008** [comment]
> Full Name : Shivam Krishna GitHub Profile Link : https://github.com/shivamkrishna1000 Participant ID : NA Approach for this Project : First I will handle the missing values if any -> use label encoding if needed -> use TfidfVectorizer for feature extraction -> Use various models such as XGB/Random Forest Classification/LR/SVM/Gradient Boosting Classifier for model training -> Model evaluation based on accuracy score. What is your participant role? SSoC Season 3
> 
> Please assign me this issue.

Implement 5-6 models for this dataset.

Assigned @shivamkrishna1000

**Comment by shivamkrishna1000** [comment]
> > Full Name : Shivam Krishna GitHub Profile Link : https://github.com/shivamkrishna1000 Participant ID : NA Approach for this Project : First I will handle the missing values if any -> use label encoding if needed -> use TfidfVectorizer for feature extraction -> Use various models such as XGB/Random Forest Classification/LR/SVM/Gradient Boosting Classifier for model training -> Model evaluation based on accuracy score. What is your participant role? SSoC Season 3
> > Please assign me this issue.
> 
> Implement 5-6 models for this dataset.
> 
> Assigned @shivamkrishna1000

Sure Sir
